packages/vite/src/node/server/transformRequest.ts

```diff
@@ -231,42 +289,100 @@ async function loadAndTransform(
   const mod = await moduleGraph.ensureEntryFromUrl(url, ssr)
   ensureWatchedFile(watcher, mod.file, root)
 
-  // transform
-  const transformStart = isDebug ? performance.now() : 0
-  const transformResult = await pluginContainer.transform(code, id, {
-    inMap: map,
-    ssr
-  })
-  const originalCode = code
-  if (
-    transformResult == null ||
-    (isObject(transformResult) && transformResult.code == null)
-  ) {
-    // no transform applied, keep code as-is
-    isDebug &&
-      debugTransform(
-        timeFrom(transformStart) + colors.dim(` [skipped] ${prettyUrl}`)
-      )
-  } else {
-    isDebug && debugTransform(`${timeFrom(transformStart)} ${prettyUrl}`)
-    code = transformResult.code!
-    map = transformResult.map
-  }
+  let result: TransformResult | null = null
 
-  if (map && mod.file) {
-    map = (typeof map === 'string' ? JSON.parse(map) : map) as SourceMap
-    if (map.mappings && !map.sourcesContent) {
-      await injectSourcesContent(map, mod.file, logger)
+  // persistent cache
+
+  const persistentCacheKey = includedInPersistentCache
+    ? (_persistentCache.getKey(id + code) ?? '') + (options.ssr ? '-ssr' : '')
+    : ''
+
+  const finalIncludedInPersistentCache =
```

Comment:
TS seems to be ok with `includedInPersistentCache` implies `_persistentCache` not null.

The idea behind me previous comment is to avoid adding a default value persistentCacheKey which can then be misused outside of a `includedInPersistentCache` block because it's effectively a string.

```ts
  const persistentCacheKey = includedInPersistentCache && ... && ...
    ? _persistentCache.getKey(id + code) + (options.ssr ? '-ssr' : '')
    : null
```

Then you can used the key as the `finalIncludedInPersistentCache` checks

Reply:
This is what I get: 
![image](https://user-images.githubusercontent.com/2798204/199305572-87631335-2410-4a4a-9052-7d6e9a3adb37.png)

Reply:
Sorry bad wording, I mean you can used the key _in place of_ the finalIncludedInPersistentCache checks. Only one variable is needed if not included -> key is null

Reply:
Yes but then I get `_persistentCache` is possibly null.